makefile

```diff
@@ -1,33 +1,35 @@
 SHELL=/bin/bash
 
 # i.e. fedora28.yaml
-ALL_TEMPLATES=$(wildcard templates/*.yaml)
+ALL_META_TEMPLATES=$(wildcard templates/templates/*.yaml)
+ALL_TEMPLATES=$(wildcard templates/*.dist.yaml)
 ALL_PRESETS=$(wildcard presets/*.yaml)
 SOURCES=$(ALL_TEMPLATES) $(ALL_PRESETS)
+METASOURCES=$(ALL_META_TEMPLATES) $(ALL_PRESETS)
 
 # i.e. fedora28
-ALL_GUESTS=$(ALL_TEMPLATES:templates/%.yaml=%)
+ALL_GUESTS=$(ALL_TEMPLATES:templates/%.dist.yaml=%)
 
 
 TEST_SYNTAX=$(ALL_GUESTS)
 TEST_UNIT=$(ALL_GUESTS)
 ifeq ($(TEST_FUNCTIONAL),ALL)
-TEST_FUNCTIONAL=fedora28 ubuntu1804 opensuse15 rhel75
+TEST_FUNCTIONAL=fedora-generic-small ubuntu-generic-small opensuse-generic-small rhel7-generic-small centos7-generic-small
 endif
 
 
 test: syntax-tests unit-tests functional-tests
 
-syntax-tests: $(TEST_SYNTAX:%=%-syntax-check)
+syntax-tests: generate $(TEST_SYNTAX:%=%-syntax-check)
 
-unit-tests: is-deployed
+unit-tests: generate is-deployed
 unit-tests: $(TEST_UNIT:%=%-apply-and-remove)
 unit-tests: $(TEST_UNIT:%=%-generated-name-apply-and-remove)
 
-functional-tests: is-deployed
+functional-tests: generate is-deployed
 functional-tests: $(TEST_FUNCTIONAL:%=%-start-wait-for-systemd-and-stop)
 
-common-templates.yaml: $(SOURCES)
+common-templates.yaml: generate $(SOURCES)
```

Comment:
For this and the trhee above:

The rule should depend on the SOURCES, and the sources should depend on the generate, as the SOURCES are build from the mETASOURCES

Reply:
That is not possible, the generated files do not exist at the beginning and SOURCES is empty.